crates/router/src/connector/bluesnap.rs

```diff
@@ -1023,18 +1023,28 @@ impl api::IncomingWebhook for Bluesnap {
             serde_urlencoded::from_bytes(request.body)
                 .into_report()
                 .change_context(errors::ConnectorError::WebhookEventTypeNotFound)?;
+        api::IncomingWebhookEvent::try_from(details)
+    }
 
-        Ok(match details.transaction_type {
-            bluesnap::BluesnapWebhookEvents::Decline
-            | bluesnap::BluesnapWebhookEvents::CcChargeFailed => {
-                api::IncomingWebhookEvent::PaymentIntentFailure
-            }
-            bluesnap::BluesnapWebhookEvents::Charge => {
-                api::IncomingWebhookEvent::PaymentIntentSuccess
-            }
-            bluesnap::BluesnapWebhookEvents::Unknown => {
-                api::IncomingWebhookEvent::EventNotSupported
-            }
+    fn get_dispute_details(
+        &self,
+        request: &api::IncomingWebhookRequestDetails<'_>,
+    ) -> CustomResult<api::disputes::DisputePayload, errors::ConnectorError> {
+        let dispute_details: bluesnap::BluesnapDisputeWebhookBody =
+            serde_urlencoded::from_bytes(request.body)
+                .into_report()
+                .change_context(errors::ConnectorError::WebhookBodyDecodingFailed)?;
+        Ok(api::disputes::DisputePayload {
+            amount: dispute_details.invoice_charge_amount,
```

Comment:
We are passing amount as higher unit to the connector like USD instead of cents, Can you confirm whether we are getting the expected value in the web hooks?

Reply:
We should always communicate amount in lower denomination irrespective of connector / flow from Hyps